src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1642,12 +1633,59 @@ FixedwingPositionControl::control_auto_takeoff(const hrt_abstime &now, const flo
 			_att_sp.pitch_body = radians(_takeoff_pitch_min.get());
 		}
 	}
+
+	if (_launch_detection_state != LAUNCHDETECTION_RES_DETECTED_ENABLEMOTORS && !_runway_takeoff.runwayTakeoffEnabled()) {
+
+		/* making sure again that the correct thrust is used,
+		 * without depending on library calls for safety reasons.
+		   the pre-takeoff throttle and the idle throttle normally map to the same parameter. */
+		_att_sp.thrust_body[0] = _param_fw_thr_idle.get();
+
+	} else if (_runway_takeoff.runwayTakeoffEnabled()) {
+
+		_att_sp.thrust_body[0] = _runway_takeoff.getThrottle(now, get_tecs_thrust());
+
+	} else {
+		/* Copy thrust and pitch values from tecs */
+		if (_landed) {
+			// when we are landed state we want the motor to spin at idle speed
+			_att_sp.thrust_body[0] = min(_param_fw_thr_idle.get(), 1.f);
+
+		} else {
+			_att_sp.thrust_body[0] = get_tecs_thrust();
+		}
+	}
+
+	// decide when to use pitch setpoint from TECS because in some cases pitch
+	// setpoint is generated by other means
+	bool use_tecs_pitch = true;
+
+	// auto runway takeoff
+	use_tecs_pitch &= !(_launch_detection_state == LAUNCHDETECTION_RES_NONE || _runway_takeoff.runwayTakeoffEnabled());
+
+	if (use_tecs_pitch) {
+		_att_sp.pitch_body = get_tecs_pitch();
+	}
```

Comment:
@tstastny I think the original logic does not set the pitch at every condition, but it was hard to know which case - I think this is why making it a separate state helps! I think once this PR is in we can start improving the logic of takeoff and landing in a more encapsulated way

Reply:
I actually don't think the extra state helps, but was just a patch because the individual conditions were left ambiguous. I'd rather each case explicitly define the intended behavior than have a catch all at the end. But to move this forward we can handle this in a subsequent PR. (with some more discussion on the logic itself / what we do / don't want to keep)